packages/idyll-cli/src/resolvers/components.js

```diff
@@ -0,0 +1,130 @@
+const p = require('path');
+const fs = require('fs');
+
+const resolve = require('resolve');
+const htmlTags = require('html-tags');
+const slash = require('slash');
+const { paramCase, pascalCase } = require('change-case');
+
+const errors = require('../errors');
+const debug = require('debug')('idyll:cli')
+
+class ComponentResolver {
+  constructor(options, paths) {
+    this.paths = paths
+    this.inputConfig = options.inputConfig;
+
+    // Set in `_loadPaths`.
+    this.componentsMap = null;
+
+    this._loadPaths();
+  }
+
+  /**
+   * Component resolution works by first constructing of Map of name -> path pairs that's later consulted in
+   * `resolve`.
+   *
+   * The Map is constructed by iterating first over the following component lists in order:
+   *  1) The default components.
+   *  2) The custom components.
+   *  3) The user-defined aliased components.
+   *
+   * Each new components list potentially overwrites previously-resolved components, according to the
+   * above order of precedence.
+   */
+  _loadPaths() {
+    var componentsMap = new Map();
+    var prioritizedDirs = [this.paths.DEFAULT_COMPONENT_DIRS, this.paths.COMPONENT_DIRS];
```

Comment:
any reason for preferring `var`s here?

Reply:
Nope, just used to writing `var` -- changed for consistency